addons/google_calendar/models/calendar.py

```diff
@@ -209,6 +216,126 @@ def _odoo_reminders_commands(self, reminders=()):
                 commands += [(0, 0, {'duration': duration, 'interval': interval, 'name': name, 'alarm_type': alarm_type})]
         return commands
 
+    def _update_future_events(self, values, time_values, recurrence_values, future_update_start):
+        """ Update future events time and info and create a new recurrence for them
+        """
+        update_dict = self._get_time_update_dict(self, time_values)
+        time_values.update(update_dict)
+
+        # Get base values from the previous recurrence.
+        [previous_recurrence_values] = self.recurrence_id.copy_data()
+
+        # Edge case: selected event is the base_event of current recurrence, save events before mass_archive.
+        is_base_event_update = (self == self.recurrence_id.base_event_id)
+        previous_recurrence_events = self.recurrence_id._get_events_from(self.start)
+        detached_events_split = self.env['calendar.event']
```

Comment:
The doc string should explain the philosophy and why this is needed and why we can't call super for google events.